src/main/java/org/broadinstitute/hellbender/tools/spark/sv/GatherSplitReadsSpark.java

```diff
@@ -44,20 +49,120 @@ protected void runTool( final JavaSparkContext ctx )
         if ( header.getSortOrder() != SAMFileHeader.SortOrder.coordinate )
             throw new GATKException("The BAM must be coordinate sorted.");
 
-        final JavaRDD<GATKRead> clusteredReads =
+        // Process the input data to produce a map of Kmer -> List of breakpoint IDs.
+        // (A breakpoint ID is just an arbitrary identifier for the putative breakpoint.)
+        // See the doc for getReadPullingMap to see a description of how this works.
+        final Broadcast<Map<Kmer,List<Long>>> broadcastMap = ctx.broadcast(getReadPullingMap(ctx));
+
+        // Process the input again, this time pulling all reads that contain kmers associated with a given breakpoint,
+        // and writing those reads into a separate FASTQ for each breakpoint.
+        getUnfilteredReads()
```

Comment:
I think it might be possible to apply some filtering here to make things more efficient -- we probably want to remove supplementary and secondary alignment records, since we should see the same read data in primary alignment records (and therefore we might actually be double-collecting these reads). Also, we might want to add in filters for duplicate reads and non-PF reads (although we shouldn't actually see the latter in Broad data).